login/service_blackbox_test.go

```diff
@@ -109,7 +110,14 @@ func (s *serviceBlackBoxTest) SetupSuite() {
 	refreshTokenSet := token.TokenSet{AccessToken: &accessToken, RefreshToken: &refreshToken}
 	s.keycloakTokenService = &DummyTokenService{tokenSet: refreshTokenSet}
 	s.osoSubscriptionManager = &testsupport.DummyOSORegistrationApp{}
-	s.Application = gormapplication.NewGormDB(s.DB, s.Configuration, factory.WithWITService(&testsupport.DevWITService{}))
+	witServiceMock := testservice.NewWITServiceMock(s.T())
+	witServiceMock.CreateUserFunc = func(p context.Context, p1 *account.Identity, p2 string) (r error) {
```

Comment:
you could move these `CreateUserFunc` and `UpdateUserFunc` methods in the `testservice.NewWITServiceMock(*testing.T)`